### ISSUE TITLE
Add Protection Realm

### ISSUE THREAD

**Issue by johnlcox** [cerner/cerner-oauth1a#3]
### Description of Issue
Add the optional `realm` parameter to errors so that it can be part of the WWW-Authenticate header.

The team owning Cerner's OAuth1a server did a spike a couple years ago and determined that the realm value should be the canonical root URI of the OAuth Server [1].

Current services acting as OAuth1a providers are setting the realm value to their own hostname, which is incorrect. Upgrading these services to use the current version of cerner-oauth1a will result in WWW-Authenticate headers that have no `realm`.

Before I open a Pull Request adding this feature, I wanted to confirm this is a desired feature.

### Expected Outcomes
- The generated WWW-Authenticate header from errors contains the `realm` parameter.

[1] https://jira.cerner.com/browse/IDMSYS-1788

**Comment by johnlcox** [comment]
In reading the OAuth1a spec I had some confusion on whether the `realm` parameter should be percent-encoded or not. I spent some time reading several specs including [rfc2617](https://tools.ietf.org/html/rfc2617) and [rfc7235](https://tools.ietf.org/html/rfc7235) which actually define the WWW-authenticate header and the `realm` parameter.

My interpretation of [section 5.4.1](https://oauth.net/core/1.0a/#auth_header) of the OAuth1a spec is that it is only the **OAuth Protocol Parameters** that should be percent-encoded, and since `realm` is defined outside of the OAuth1a spec it should not be encoded.

I also posted a StackOverflow question that I answered myself with additional details on my reasoning that `realm` should not be percent-encoded: https://stackoverflow.com/questions/54406117/should-the-www-authenticate-realm-parameter-be-encoded

Based on this interpretation I plan to implement this with special handling of the `realm` parameter to not be encoded with the other OAuth parameters.
